osdep/path-unix.c

```diff
@@ -65,6 +66,39 @@ static void path_init(void)
         err = err || MKPATH(mpv_state, "%s/.local/state/mpv", home);
     }
 
+    char xdg_user_dirs[CONF_MAX];
+    if (xdg_config && xdg_config[0]) {
+        err = err || MKPATH(xdg_user_dirs, "%s/user-dirs.dirs", xdg_config);
+    } else if (home && home[0]) {
+        err = err || MKPATH(xdg_user_dirs, "%s/.config/user-dirs.dirs", home);
+    }
+
+    // Attempt to read user-dirs for XDG_DESKTOP_DIR
+    if (mp_path_exists(xdg_user_dirs)) {
+        char line[4096];
+        FILE *user_dirs = fopen(xdg_user_dirs, "r");
+        while (fgets(line, sizeof(line), user_dirs)) {
```

Comment:
Could we write it in mpv style? Using bstr and all goodies? You can look at `m_config_parse` as an good example how to parse text files.

Reply:
I converted every line to bstr and went from there. I can't read the entire stream like m_config_parse does though.